src/main/java/io/ddavison/conductor/listeners/Watchman.java

```diff
@@ -0,0 +1,39 @@
+package io.ddavison.conductor.listeners;
+
+import io.ddavison.conductor.Locomotive;
+import io.ddavison.conductor.util.ScreenShotUtil;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+public abstract class Watchman extends TestWatcher {
+
+    public abstract Locomotive getLocomotive();
+    public abstract void quit();
+
+    private boolean failure;
+    private Throwable e;
+
+    @Override
+    protected void failed(Throwable e, Description description) {
+        if (getLocomotive().configuration.isScreenshotOnFail()) {
+            failure = true;
```

Comment:
Why are we setting a flag here and not taking a screenshot directly here? The call to `quit()` could stay in `finished()` if needed.

Reply:
That's a good question, I forget way, I know there was a reason. Let me check if I can move it.